### DISCUSSION TITLE
Local Twill install

### DISCUSSION THREAD

**Discussion by vbulash** [area17/twill#1113] [Q&A]
Is it possible to use Twill in local development environment - no Homestead / Vagrant, just internal PHP server (php artisan serve)

**Comment by pboivin** [comment]
Hi @vbulash. Yes, this is possible. You will need a local DB server and possibly to configure your `/etc/hosts`. A quick example:

**add to file: /etc/hosts**
```
127.0.0.1  example.test
```

**update file: .env**
```
APP_PORT=8000
APP_URL=http://example.test
ADMIN_APP_URL=http://example.test
ADMIN_APP_PATH=admin
```

**then run:**
```
php artisan serve --host=example.test --port=8000
```

**Comment by vbulash** [comment]
Sorry, seems not working.

Of course, php artisan twill:install made before all steps.
Then step:

/etc/hosts:

127.0.0.1	neurolife.test

.env:
APP_NAME=Neurolife
APP_ENV=local
APP_KEY=base64:...
APP_DEBUG=true
APP_URL=http://neurolife.test

# Twill
APP_PORT=8000
ADMIN_APP_URL="{$APP_URL}"
ADMIN_APP_PATH=admin
FILE_LIBRARY_ENDPOINT_TYPE=local
FILE_LIBRARY_LOCAL_PATH=uploads/
MEDIA_LIBRARY_ENDPOINT_TYPE=local
MEDIA_LIBRARY_LOCAL_PATH=uploads
MEDIA_LIBRARY_IMAGE_SERVICE=A17\Twill\Services\MediaLibrary\Glide
Start server:
> php artisan serve —host=neurolife.test --port=8000
config/twill.php:

<?php

return [
    'locale' => 'ru',
    'fallback_locale' => 'en',

    'enabled' => [
        'users-management' => true,
        'media-library' => true,
        'file-library' => true,
        'block-editor' => true,
        'buckets' => true,
        'users-image' => false,
        'settings' => true,
        'dashboard' => true,
        'search' => true,
        'users-description' => false,
        'activitylog' => true,
        'users-2fa' => false,
        'users-oauth' => false,
    ],
];

Module ‘blog’ added by artisan.

config/twill-navigation.php:
<?php

return [
    'blogs' => [
        'title' => 'Блог',
        'module' => true
    ]
];
resources/views/admin/blogs/form.blade.php:

@extends('twill::layouts.form')

@section('contentFields')
    @formField('input', [
    'name' => 'title',
    'label' => 'Заголовок',
    'placeholder' => 'Введите заголовок статьи блога',
    'maxlength' => 100
    ])

    @formField('wysiwyg', [
    'name' => 'description',
    'label' => 'Описние',
    'toolbarOptions' => [
        [ 'header' => [1, 2, false] ],
        'clean', 'bold', 'link', 'italic', 'list-ordered', 'list-unordered', [
        'indent' => '-1'], [ 'indent' => '+1' ]
    ],
    'placeholder' => 'Введите описание статьи блога',
    'maxlength' => 200,
    'editSource' => true,
    ])
@stop
That’s it.
Try open http://neurolife.test:8000 <http://neurolife.test:8000/> - ok, initial Laravel page
Try open http://admin.neurolife.test:8000 <http://admin.neurolife.test:8000/> or http://neurolife.test:8000/admin <http://neurolife.test:8000/admin> - 404 error.
So, twill not installed or not configured properly.
What’s up?

**Reply by celsojr-websters** [reply]
Have you tried Docker?

**Reply by vbulash** [reply]
No
